### PR TITLE
task 5 to 7

### PR DESCRIPTION
**Pull request by StevenJune**
Task 5 to 7

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -74,3 +74,193 @@ describe("4. GET /api/articles/:article_id", () => {
...
+          votes: 200,
+          author: "rogersop",
+          body: "Bastet walks amongst us, and the cats are taking arms!",
+          created_at: "2020-08-03T13:14:00.000Z",
+          topic: "cats",
+        });
+      });
+  });
+
+  it("status:201, responds with invalid post key", () => {
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js]: This should be a 400

File: __tests__/app.test.js [hunk 2]
```diff
@@ -74,3 +74,193 @@ describe("4. GET /api/articles/:article_id", () => {
...
+
+   test("status:400, patch article_id does not exist in table", () => {
+    const article_id = 999;
+    const votesCount = {
+        inc_votes: 200,
+      };
+    return request(app)
+      .patch(`/api/articles/${article_id}`)
+      .send(votesCount)
+       .expect(400)
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js]: This should be a 404 as the article id is a number but doesn't exist

File: __tests__/app.test.js [hunk 3]
```diff
@@ -74,3 +74,193 @@ describe("4. GET /api/articles/:article_id", () => {
...
+      .then(({ body }) => {
+        expect(body.msg).toBe(`No article found for updating votes in this article_id : ${article_id}`);
+      });
+  });
+
+  test.only("status:400, invalid article_id feed in", () => {
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js]: Don't forget to remove the .only from your tests!

File: __tests__/app.test.js [hunk 4]
```diff
@@ -74,3 +74,193 @@ describe("4. GET /api/articles/:article_id", () => {
...
+      .patch(`/api/articles/${article_id}`)
+      .send(votesCount)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe(
+          `invalid input syntax for type integer: "${article_id}"`
+        );
+      });
+  });
+
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js:167]: You could also add 1 more test here. What if inc_votes is not a number?

File: __tests__/app.test.js [hunk 5]
```diff
@@ -74,3 +74,193 @@ describe("4. GET /api/articles/:article_id", () => {
...
+});
+
+
+describe("7. GET /api/articles/:article_id/comment", () => {
+    test("status:200, responds with a single matching article with comment count", () => {
+      const article_id = 3;
+      return request(app)
+        .get(`/api/articles/${article_id}/comment`)
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js]: This endpoint url is incorrect. It should just be an update to your existing /api/articles/:article_id

File: app.js [hunk 6]
```diff
@@ -1,15 +1,23 @@
...
+        getArticlesByIdComment,getArticlesByTopic } = require("./controllers/news");
 
 const app = express();
 
 app.get("/api", getApi);
 app.get("/api/topics", getTopics);
 app.get("/api/articles/:article_id", getArticlesById);
+app.get("/api/users",getUsers);
+app.patch("/api/articles/:article_id", patchArticlesById);
+app.get("/api/articles/:article_id/comment", getArticlesByIdComment);
```
Comment (Rayhaan2k9) [inline app.js]: This endpoint url is incorrect. You don't need to create a new endpoint, you just need to add some code to your existing /api/articles/:article_id

**Review by Rayhaan2k9 — CHANGES_REQUESTED** [review]
A few things that need to be fixed Steven. I have left comments for you. The main thing is ticket 7. This doesn't need to be a new endpoint. It is /api/articles/:article_id. You can just update your code in that endpoint for ticket 7 :)

File: app.js [hunk 7]
```diff
@@ -8,16 +8,16 @@ const app = express();
 
 app.get("/api", getApi);
 app.get("/api/topics", getTopics);
-app.get("/api/articles/:article_id", getArticlesById);
+//app.get("/api/articles/:article_id", getArticlesById);  is going to be replaced by function getArticlesByIdComment
```
Comment (leekli) [inline app.js]: Please remove this commented out code

File: app.js [hunk 8]
```diff
@@ -8,16 +8,16 @@ const app = express();
...
 app.get("/api/users",getUsers);
 app.patch("/api/articles/:article_id", patchArticlesById);
-app.get("/api/articles/:article_id/comment", getArticlesByIdComment);
-app.get("/api/articles?topic=:topic", getArticlesByTopic);
+app.get("/api/articles/:article_id", getArticlesByIdComment);
+//app.get("/api/articles?topic=:topic", getArticlesByTopic);
```
Comment (leekli) [inline app.js]: Please remove this commented out code

File: app.js [hunk 9]
```diff
@@ -8,16 +8,16 @@ const app = express();
...
+//app.get("/api/articles", getArticlesByTopic);
```
Comment (leekli) [inline app.js]: Please remove this commented out code

File: app.js [hunk 10]
```diff
@@ -26,12 +26,12 @@ app.use((err, req, res, next) => {
   if (err.code === "22P02") {
     res.status(400).send({ msg: err.message || "Bad Request" });
   } else if (err.status && err.msg) {
-    console.log('aaaaaa');
     res.status(err.status).send({ msg: err.msg });
   }
   // handle specific psql errors
   else {
     // respond with an internal server error
+    console.log(err);
```
Comment (leekli) [inline app.js]: This console log does not need to be here

File: controllers/news.js [hunk 11]
```diff
@@ -39,54 +41,42 @@ exports.getUsers = (req, res, next) => {
...
+};
 
+exports.getArticlesByTopic = (req, res, next) => {
+  const myArg = req.query.topic;
+  searchArticlesByTopic(myArg)
+    .then((articles) => {
+      res.status(200).send({ articles });
+    })
+    .catch(next);
+};
```
Comment (leekli) [inline controllers/news.js]: Steven - What ticket is this function for? It's not in tickets 5-7 so I am not sure why you have this function here? There is not an endpoint for it in your app.js either. Are you trying to start ticket 8 here? If so, you shouldn't be working on ticket 8 until tickets 5-7 have been approved.

File: models/news.js [hunk 12]
```diff
@@ -89,7 +90,4 @@ exports.searchArticlesByTopic = (topic) => {
     }
     return article;
   });
-
 };
-
-
```
Comment (leekli) [inline models/news.js]: Steven - What ticket is this function for? It's not in tickets 5-7 so I am not sure why you have this function here? There is not an endpoint for it in your app.js either. Are you trying to start ticket 8 here? If so, you shouldn't be working on ticket 8 until tickets 5-7 have been approved.

**Review by leekli — CHANGES_REQUESTED** [review]
Hi Steven, this pull request still is not quite finished.

There are some comments attached I would like you to look at:
* There is some commented out code and a console log which needs removing 
* I can see functions for 'searchingArticlesByTopic' - I am not sure why these are here? they are not part of tickets 5-7? Why are these here? If it's because ticket 8 asks for articles by topic, then you should not be working on ticket 8 yet, you only do ticket 8 tasks once and only once tickets 5-7 have been approved. Please can you remove any search articles by topic functions and tests.

**Review by leekli — APPROVED** [review]
Thank you, Steve.

You can now MERGE this branch.

Please now move on to ticket number 8 only, please only work on 1 ticket at a time in each branch, please do not try to complete ticket 9 at the same time. If you have any questions, then please ask us.

**Pull request merged**